File: Math/All factors of a given Number/All factors of a given Number.cpp
//We include de I/O library
#include <iostream>
/* We include the stdenteredNumeramespace so we don't have to write std::
Everytime we want to use "cout" or "cin" */
using namespace std;

int main()
{
//    We declare our variables
    int enteredNumer, i, factorization;
    
//    Ask for a number
    cout << "Enter a number of your choice (positive and integeer): ";
//    Assing the entered value to the "enteredNumer" variable
    cin >> enteredNumer;
    
    cout << "These are the factors for " << enteredNumer << ": " << endl;
    
    factorization = enteredNumer;
    
//    Every number has 1 as it's factor, so we'll say this right away
    cout << 1 << endl;
    
//    Check the factors (starting with 2, until the number is reached)
    for(i = 2; i <= enteredNumer; ++i)
    {
//        If the divided number has 0 as the division reminder, it'll be a factor
        
        while (factorization % i == 0){
//            Print this number everytime
            cout << i << endl;
            
//            As it is a factor, let's divide by it to find the next one!
            factorization = factorization / i;
        };
    }
    
    return 0;
}

File: Math/All factors of a given Number/Python/All factors of a given Number.py
"""
         Print all factors of a given number

"""

#Enter your number
num = input("Enter a number: ");
#initialize your count as 0
count = 0;
number = int(num);
#run a loop to find factors
#Check the factors (starting with 2, until the number is reached)
for i in range(2, number-1):
    #If the divided number has 0 as the division reminder, it'll be a factor
    if number%i == 0:
        #Print this number everytime
        print(i);
        i += 1;
        count += 1;
#If number is not divisible by any number other than 1 and number itself,then it is prime
if count==0:
    print(number,"is a prime number.");
